pages/interface/components/Content/index.js

```diff
@@ -430,7 +430,21 @@ function EditMode({ contentObject, setContentObject, setComponentMode, localStor
 
         if (response.status >= 401) {
           return (responseBody) => {
-            setGlobalErrorMessage(createErrorMessage(responseBody));
+            if (responseBody.error_location_code === 'MODEL:CONTENT:CREDIT_OR_DEBIT_TABCOINS:NEGATIVE_USER_EARNINGS') {
+              setGlobalErrorMessage(
+                <>
+                  <Text>
+                    {responseBody.message} {responseBody.action}
```

Comment:
> > Será que não é interessante deixar isso dentro de `createErrorMessage`?
> 
> Transferir esse `if` para lá? Não acho que faz sentido porque o `createErrorMessage` é genérico, enquanto que esse erro é uma resposta específica para essa ação.

Acho que faz sentido manter centralizado onde traduzimos as mensagens de erro recebidas da API para algo mais adequado à UI, mas ainda falta criar um componente responsável pela renderização, então tudo bem deixar isso para depois.

Mas dá para continuar usando o `createErrorMessage` para retornar o texto:

```suggestion
                    {createErrorMessage(responseBody, { omitErrorId: true })}
```

Reply:
Vou realizar as alterações. Me parece que colocar `omitErrorId` aqui não é o melhor, porque omitiria o ID de outros erros. Veja se minha proposta faz sentido.

Reply:
> Me parece que colocar `omitErrorId` aqui não é o melhor, porque omitiria o ID de outros erros.

Não entendi porquê omitiria o ID de outros erros se `{createErrorMessage(responseBody, { omitErrorId: true })}` estava dentro da condição `responseBody.error_location_code === 'MODEL:CONTENT:CREDIT_OR_DEBIT_TABCOINS:NEGATIVE_USER_EARNINGS'`.

Mas seguir essa sugestão ao pé da letra também não resolve o que está mais estranho, que é salvar um componente React usando `useState`. Ou não vê problema nisso?

> Veja se minha proposta faz sentido.

Na última proposta me parece ser ao pé da letra o que eu sugeri primeiro, pois está lidando totalmente dentro de `createErrorMessage`, mas isso transforma a função em um componente React, então precisaria mudar ela para outra pasta e deixar a função em `PascalCase`, o que talvez implicaria mudar todos os locais que usam a `createErrorMessage`. Pode não valer a pena.

Para não precisa revolucionar, dá para deixar a `createErrorMessage` da forma que está, lidando apenas com o texto, e criar um componente de UI. Esse componente pode até ser criado dentro de `Content` e não precisa ser algo muito flexível por enquanto. E deve compensar usar a `createErrorMessage` nesse componente, mas veja o que acha melhor, evitando fugir das convenções do React.

Reply:
Entendi errado a sua sugestão; agora deve estar mais próximo do esperado.

Modifiquei os `setGlobalErrorMessage` para permitir passar também o `omitErrorId`. Outra opção seria aceitar a resposta "flat" (`setGlobalErrorMessage(responseBody)`) e modificar apenas onde o `omitErrorId` fosse definido, por exemplo:

```jsx
setGlobalErrorMessage({ ...responseBody, omitErrorId: true });
```